samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,293 @@
+            "ca-path", G_TYPE_STRING, ca_cert_path,
+            "role-aliases", G_TYPE_STRING, role_alias, NULL);
+    
+    g_object_set(G_OBJECT (kvssink), "iot-certificate", iot_credentials, NULL);
+        gst_structure_free(iot_credentials);
+    // kvssink will search for long term credentials in envvar automatically so no need to include here
+    // if no long credentials or IoT credentials provided will look for credential file as last resort.
+    } else if(nullptr != (credential_path = getenv("AWS_CREDENTIAL_PATH"))){
+        g_object_set(G_OBJECT (kvssink), "credential-path", credential_path, NULL);
+    }
+}
+
+// This function handles the intermittent starting and stopping of the stream in a loop.
+void stopStartLoop(GstElement *pipeline) {
+    std::mutex cv_m;
+    std::unique_lock<std::mutex> lck(cv_m);
+
+    while (!terminated) {
+        // Using cv.wait_for to break sleep early upon signal interrupt.
+        if (cv.wait_for(lck, std::chrono::seconds(KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS)) != std::cv_status::timeout) {
+            break;
+        }
+
+        LOG_INFO("[KVS sample] Stopping stream to KVS for " << KVS_INTERMITTENT_PAUSED_INTERVAL_SECONDS << " seconds");
+
+        // EOS event pushes frames buffered by the h264enc element down to kvssink.
+        GstEvent* eos = gst_event_new_eos();
+        gst_element_send_event(pipeline, eos);
+
+        // Wait for the EOS event to return from kvssink to the bus which means all elements are done handling the EOS.
+        // We don't want to flush until the EOS is done to ensure all frames buffered in the pipeline have been processed.
+        cv.wait(lck);
+
+        // Flushing to remove EOS status.
+        GstEvent* flush_start = gst_event_new_flush_start();
+        gst_element_send_event(pipeline, flush_start);
+
+        // Using cv.wait_for to break sleep early upon signal interrupt. Checking for termination again before waiting.
+        if (terminated || cv.wait_for(lck, std::chrono::seconds(KVS_INTERMITTENT_PAUSED_INTERVAL_SECONDS)) != std::cv_status::timeout) {
+            break;
+        }
+
+        LOG_INFO("[KVS sample] Starting stream to KVS for " << KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS << " seconds");
+        GstEvent* flush_stop = gst_event_new_flush_stop(true);
+        gst_element_send_event(pipeline, flush_stop);
+    }
+    LOG_DEBUG("[KVS sample] Exited stopStartLoop");
+}
+
+int main (int argc, char *argv[])
+{
+    signal(SIGINT, sigint_handler);
+
+    CustomData customData;
+    GstElement *pipeline, *source, *clock_overlay, *video_convert, *source_filter, *encoder, *sink_filter, *kvssink;
+    GstCaps *source_caps, *sink_caps;
+    GstBus *bus;
+    guint bus_watch_id;
+    StreamSource source_type;
+    char stream_name[MAX_STREAM_NAME_LEN + 1];
+
+    gst_init (&argc, &argv);
+
+
+    /* Parse input arguments */
+
+    // Check for invalid argument count.
+    if (argc > 3 || argc < 2) {
```

Comment:
Would suggest splitting this into 2 checks to clarify what is missing.

Reply:
Actually, opted for allowing for no extra args to use kvssink default stream name.